src/Serializable.ts

```diff
@@ -8,19 +8,41 @@ import { SerializableBase } from "./SerializableBase"
 export class Serializable extends SerializableBase implements ISerializable {
     public static fromUnknown(value: any): Serializable {
         const obj: any = value
+        let type
         if (obj["@type"]) {
-            const result = SerializableBase.getModule(obj["@type"])
-            if (!result) {
-                throw new ServalError(
-                    `Type '${obj["@type"]}' was not found within reflection classes. You might have to install a module first.`
-                )
+            if (typeof obj["@type"] !== "string") {
+                throw new ServalError("Type is not a string.")
             }
-            if (typeof result.fromJSON === "function") {
-                return result.fromJSON(value)
+            type = `${obj["@type"]}`.trim()
+        }
+
+        let version = 1
+        if (obj["@version"]) {
+            try {
+                version = parseInt(obj["@version"])
+            } catch (e) {
+                throw new ServalError("Version is not a number.")
             }
-            return result.from(value, result)
         }
-        return Serializable.from(value)
+
+        if (!type) {
+            return Serializable.from(value)
+        }
+
+        if (!type && version) {
+            throw new ServalError("Version is set for an unknown type.")
+        }
+
+        const result = SerializableBase.getModule(type, version)
+        if (!result) {
+            throw new ServalError(
+                `Type '${type}' with version ${version} was not found within reflection classes. You might have to install a module first.`
```

Comment:
Ich überlege gerade, ob diese Fehlermeldung nicht verwirrend ist, wenn man keine Version angegeben hat. Weil dann steht in der Fehlermeldung was von "version 1".

Reply:
Finde ich sogar sehr gut, man muss halt verstehen (und das muss man eh), dass keine Versionsangabe => Version 1 ist. So ist man sich sicher, dass wenn ein Fehler auftritt, auch eine Version in der Fehlermeldung ist.